### PR TITLE
add MobileNetV1 model

### PR DESCRIPTION
**Pull request by hollance**
# What does this PR do?

Adds the MobileNet V1 model to the library.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by hollance** [comment]
Note: The checkpoints still point to my own account, but should be changed to `google` once the changes are approved.

File: docs/source/en/model_doc/mobilenet_v1.mdx
Comment (sgugger) [inline docs/source/en/model_doc/mobilenet_v1.mdx:43]: Maybe we can start experimenting with a backbone feature to return those in a `MobileNetV1Backbone`?

File: docs/source/en/model_doc/mobilenet_v1.mdx
Comment (sgugger) [inline docs/source/en/model_doc/mobilenet_v1.mdx:47]: For consistency with other models having a version in their name, I'd use a lowercased v (like `LayoutLMv2`) in all class names.

File: src/transformers/models/mobilenet_v1/configuration_mobilenet_v1.py
Comment (sgugger) [inline src/transformers/models/mobilenet_v1/configuration_mobilenet_v1.py:31]: You know but flagging in a comment those will need to be updated :-)

File: src/transformers/models/mobilenet_v1/configuration_mobilenet_v1.py
Comment (sgugger) [inline src/transformers/models/mobilenet_v1/configuration_mobilenet_v1.py:41]: Here too.

File: src/transformers/models/mobilenet_v1/modeling_mobilenet_v1.py
Comment (sgugger) [inline src/transformers/models/mobilenet_v1/modeling_mobilenet_v1.py:54]: Nit: this function and the second one would go better in the conversion module IMO.

**Review by sgugger — APPROVED** [review]
Very clean! Thanks a lot for adding this new model!

File: src/transformers/models/mobilenet_v1/modeling_mobilenet_v1.py
Comment (sgugger) [reply src/transformers/models/mobilenet_v1/modeling_mobilenet_v1.py:54]: Everyone always copies it from BERT, but it's to load a TensorFlow 1 checkpoint inside a PyTorch model only. So I'm not sure it applies here. The checkpoints you have are TF2 checkpoints, no?

Comment (hollance) [reply src/transformers/models/mobilenet_v1/modeling_mobilenet_v1.py:54]: They're TF1 or at least the compat.v1 stuff. These models were originally implemented with the old "slim" wrapper that Google used for a while.

File: docs/source/en/model_doc/mobilenet_v1.mdx
Comment (sgugger) [reply docs/source/en/model_doc/mobilenet_v1.mdx:47]: True indeed, forgot this one. I prefer lowercase v, but it's your call!

File: src/transformers/models/mobilenet_v1/modeling_mobilenet_v1.py
Comment (sgugger) [reply src/transformers/models/mobilenet_v1/modeling_mobilenet_v1.py:54]: In this case, the function will need to stay here indeed!

File: docs/source/en/model_doc/mobilenet_v1.mdx
Comment (hollance) [reply docs/source/en/model_doc/mobilenet_v1.mdx:47]: It kinda depends on the letter before it: `LayoutLMV2` makes it seem the V belongs to LMV, so `v2` is better there. But `Debertav2` looks worse than `DebertaV2`. Personally, I prefer `MobileNetV1` over `MobileNetv1`.

File: src/transformers/models/mobilenet_v1/feature_extraction_mobilenet_v1.py
Comment (alaradirik) [inline src/transformers/models/mobilenet_v1/feature_extraction_mobilenet_v1.py:141]: Is `crop_size` necessary over here? Cropping input images might lead to worse classification performance.

**Review by alaradirik — APPROVED** [review]
Hey @hollance, thank you for adding this! The PR seems to be in good shape, and it'd be great to rebase this branch to main and merge the PR.

@sgugger @NielsRogge what do you think?

File: src/transformers/models/mobilenet_v1/feature_extraction_mobilenet_v1.py
Comment (hollance) [reply src/transformers/models/mobilenet_v1/feature_extraction_mobilenet_v1.py:141]: These models are trained on 224x224 images. Using larger images may not work that well since it probably wasn't trained on multi-scale inputs. Plus, it will be slower and the point of these models is speed.

**Comment by hollance** [comment]
I've rebased it so that it can merged. Perhaps @sgugger or @NielsRogge could merge it?

File: src/transformers/models/mobilenet_v1/configuration_mobilenet_v1.py
Comment (sgugger) [inline src/transformers/models/mobilenet_v1/configuration_mobilenet_v1.py:31]:
```suggestion
    "google/mobilenet_v1_1.0_224": "https://huggingface.co/google/mobilenet_v1_1.0_224/resolve/main/config.json",
    "google/mobilenet_v1_0.75_192": "https://huggingface.co/google/mobilenet_v1_0.75_192/resolve/main/config.json",
```

File: src/transformers/models/mobilenet_v1/configuration_mobilenet_v1.py
Comment (sgugger) [inline src/transformers/models/mobilenet_v1/configuration_mobilenet_v1.py:41]: Should use `google` here too.

**Review by sgugger — APPROVED** [review]
Just a couple of checkpoints to fix and we can merge this, thanks!

File: src/transformers/models/mobilenet_v1/configuration_mobilenet_v1.py
Comment (hollance) [reply src/transformers/models/mobilenet_v1/configuration_mobilenet_v1.py]: Whoops, thanks for catching that. I noticed that I made the same mistake in the MobileNetV2 config, so I've included that in this PR too (it's only a small change).

**Pull request merged**
